openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -586,6 +735,23 @@ def delete_cosmetic_attribute(self, attr_name):
         delattr(self, '_'+attr_name)
         self._cosmetic_attribs.remove(attr_name)
 
+    @staticmethod
+    def _split_attribute_index(item):
+        """Split the attribute name from the final index.
+
+        For example, the method takes 'k2' and returns the tuple ('k', 1).
+        If attribute_name doesn't end with an integer, it returns (item, None).
+        """
+        # Match any number (\d+) at the end of the string ($).
+        match = re.search(r'\d+$', item)
```

Comment:
I agree that this is the best way to do it. Referencing this line in #338 to keep track of this behavior.